src/main/java/de/tum/in/www1/artemis/domain/lecture/Slide.java

```diff
@@ -80,35 +86,28 @@ public void onLoad() {
         prevSlideImagePath = slideImagePath; // save current path as old path (needed to know old path in onUpdate() and onDelete())
     }
 
-    @PrePersist
-    public void beforeCreate() {
-        var targetFolder = Path.of(FilePathService.getAttachmentUnitFilePath(), getAttachmentUnit().getId().toString(), "slide", String.valueOf(getSlideNumber())).toString();
-        slideImagePath = fileService.manageFilesForUpdatedFilePath(prevSlideImagePath, slideImagePath, targetFolder, (long) getSlideNumber(), false);
-    }
-
     /**
-     * Will be called after the entity is persisted (saved).
-     * Manages files by taking care of file system changes for this entity.
+     * Before persisting the slide, we need to move the file from the temp folder to the actual folder
      */
-    @PostPersist
-    public void afterCreate() {
-        // replace placeholder with actual id if necessary (id is no longer null at this point)
-        if (slideImagePath != null && slideImagePath.contains(Constants.FILEPATH_ID_PLACEHOLDER)) {
-            slideImagePath = slideImagePath.replace(Constants.FILEPATH_ID_PLACEHOLDER, getAttachmentUnit().getId().toString());
+    @PrePersist
+    public void beforeCreate() {
+        if (slideImagePath == null) {
+            return;
         }
+        Path targetFolder = FilePathService.getAttachmentUnitFilePath().resolve(Path.of(getAttachmentUnit().getId().toString(), "slide", String.valueOf(getSlideNumber())));
+        slideImagePath = entityFileService.moveFileBeforeEntityPersistenceWithIdIfIsTemp(slideImagePath, targetFolder, false, (long) getSlideNumber());
     }
 
     @PreUpdate
     public void onUpdate() {
-        // move file and delete old file if necessary
-        var targetFolder = Path.of(FilePathService.getAttachmentUnitFilePath(), getAttachmentUnit().getId().toString(), "slide", String.valueOf(getSlideNumber())).toString();
-        slideImagePath = fileService.manageFilesForUpdatedFilePath(prevSlideImagePath, slideImagePath, targetFolder, (long) getSlideNumber(), false);
+        Path targetFolder = FilePathService.getAttachmentUnitFilePath().resolve(Path.of(getAttachmentUnit().getId().toString(), "slide", String.valueOf(getSlideNumber())));
```

Comment:
Consider putting the targetFolder generation in a function (as it avoids code duplication with beforeCreate()).